src/util.rs

```diff
@@ -0,0 +1,47 @@
+use std::collections::HashMap;
+
+use matrix_sdk::{
+    ruma::{OwnedRoomAliasId, OwnedRoomId},
+    Client,
+};
+
+pub(super) struct Util {
+    client: Client,
+    /// In-memory cache for the room alias to room id mapping.
+    room_cache: HashMap<OwnedRoomAliasId, OwnedRoomId>,
+}
+
+impl Util {
+    pub fn new(client: Client) -> Self {
+        Self {
+            client,
+            room_cache: Default::default(),
+        }
+    }
+
+    pub fn resolve_room(&mut self, room: &str) -> anyhow::Result<String> {
```

Comment:
I'd slightly prefer if this now returned `anyhow::Result<Option<String>>`, to distinguish the case of "we couldn't resolve an apparently correctly looking room alias" (`Err(...)`) from "no way the room string indicates a room alias" (`Ok(None)`)

Reply:
I've gone ahead and made this change, though personally I feel it makes more sense that this function just returns a `Result`. Either the room can be resolved or it can't.

1. Do you have a use case in mind where we'd want different behaviour based on the room appearing valid or not?
2. If yes, then what should the test be for this? I went with `room.starts_with("#")`, but that feels kind of arbitrary. What if the user meant to pass in a room id, but forgot the leading `#`?

Overall, I think trying to guess user intent in this function could end up leading to mistakes. That said, I'm happy to leave this as is :).

Reply:
Yeah, the test should be start with `#` (likely a room alias) or `!` (likely a room id), I'll tweak it. I think it's nice to report to the user when they try to enter a room id but it doesn't match either pattern; that might indicate they've pasted a room id incorrectly, or something like that.

Reply:
it would be nice to test all of this; could be doable with a small refactoring of `RoomResolver` into a trait, so we can have an actual impl in matrix, and test impls for the tests. Some other day :)